eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/implementation/AmqpReceiveLink.java

```diff
@@ -33,4 +34,18 @@
      * @param credits Number of credits to add to the receive link.
      */
     void addCredits(int credits);
+
+    /**
+     * Gets the current number of credits this link has.
+     *
+     * @return The number of credits (deliveries) this link has.
+     */
+    int getCredits();
+
+    /**
+     * Adds an event listener that is invoked when there are no credits on the link left. Informs the
+     *
+     * @param creditSupplier Supplier that returns the number of credits to add to the link.
+     */
+    void setEmptyCreditListener(Supplier<Integer> creditSupplier);
```

Comment:
A little confused about this. It seems like you can only have one listener, despite the 'Adds an event listener' JavaDoc.